### PR TITLE
View federated API details

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-4195

## Description

PR with changes required to view and update federated API details without any issue.
Documentation tab is now enabled which lets user to create documentation and add metadata to the federated API.
Updating federated API is now migrated to the new architecture style.  

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-thmojfbtxr.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/ApiMapper.java [hunk 1]
```diff
@@ -182,6 +168,15 @@ io.gravitee.rest.api.management.v2.rest.model.ApiV1 mapToV1(
     @Mapping(target = "id", expression = "java(apiId)")
     UpdateApiEntity map(UpdateApiV4 updateApi, String apiId);
 
+    @Mapping(target = "id", expression = "java(apiId)")
+    UpdateApiEntity map(UpdateApiFederated updateApi, String apiId);
+
+    @Mapping(source = "updateApiFederated.lifecycleState", target = "apiLifecycleState")
+    @Mapping(source = "updateApiFederated.apiVersion", target = "version")
+    @Mapping(expression = "java(apiId)", target = "id")
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/ApiMapper.java]:
```suggestion
@Mapping(source = "apiId", target = "id")
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource_UpdateApiTest.java [hunk 2]
```diff
@@ -190,4 +199,81 @@ public void should_return_bad_request_when_updating_v4_api_with_v2_defition() {
...
+
+    void primaryOwnerInit() {
+        final var API_NAME = "my-api";
+        final var ROLE_ID = "role-id";
+        Map.Entry<String, PrimaryOwnerEntity> entry = Map.entry(
+            API_NAME,
+            PrimaryOwnerEntity.builder().id(API_NAME).type(PrimaryOwnerEntity.Type.USER).build()
+        );
+
+        roleQueryService.initWith(
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource_UpdateApiTest.java]: you can use `roleQueryService.resetSystemRoles(ORGANIZATION)`

Comment (JedrzejJanasiak) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiResource_UpdateApiTest.java]: right, done :)

**Comment by JedrzejJanasiak** [comment]
Update: Fixed all places with wildcard imports

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**